core/src/main/scala/org/apache/spark/TaskContextImpl.scala

```diff
@@ -54,6 +54,7 @@ private[spark] class TaskContextImpl(
     @transient private val metricsSystem: MetricsSystem,
     // The default value is only used in tests.
     override val taskMetrics: TaskMetrics = TaskMetrics.empty,
+    override val cpus: Int = 0,
```

Comment:
Default to CPUS_PER_TASK ?

Reply:
I think I have similar comment here, do we really need a default here?  Perhaps more tests here than below.  Also I don't think this has access to CPUS_PER_TASK

Reply:
The comment said The default value is only used in tests. I changed to 1. If there is no default, we need to change a lot of tests.

Reply:
CPUS_PER_TASK is simply "spark.task.cpus" with default 1 ... which should be available from `SparkEnv.conf` ?
`SparkEnv.conf.get(config.CPUS_PER_TASK)` ?